akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala

```diff
@@ -290,6 +294,12 @@ final class ClusterShardingSettings(
   def withStateStoreMode(stateStoreMode: ClusterShardingSettings.StateStoreMode): ClusterShardingSettings =
     copy(stateStoreMode = stateStoreMode)
 
+  def withPassivateIdleEntitiesAfter(duration: Duration): ClusterShardingSettings =
+    copy(passivateIdleEntityAfter = duration)
+
+  def withPassivateIdleEntityAfter(duration: java.time.Duration): ClusterShardingSettings =
+    copy(passivateIdleEntityAfter = duration.asScala)
```

Comment:
Duration vs FiniteDuration?

Reply:
Yes, had bad luck when I was thinking about how to represent disabled auto-passivation ofc 0 is a fine `FiniteDuration` as well

Reply:
Another aspect is that I would like that it is enabled by default for Typed Sharding, but we can't just do that for Untyped Sharding in a patch release.

Or we just leave it off for everything in Akka 2.5 and enable it for all in 2.6?

Reply:
I was thinking that it is impossible to decide on a reasonable default for all systems and that you have to know what the entities does and how expensive starting them up is etc. so that's why I left it as disabled by default.

Reply:
I think a decent default is a few minutes, let's say 5. The more common consequence is that users don't care/know about passivation and then resulting in all entities always being in memory.

Reply:
Ok, then I think we should leave it off, and introduce the 5 min default for all in 2.6